validator/src/journal/block_manager.rs

```diff
@@ -398,13 +402,152 @@ impl BlockManagerState {
 #[derive(Default, Clone)]
 pub struct BlockManager {
     state: Arc<BlockManagerState>,
+    persist_lock: Arc<RwLock<()>>,
 }
 
 impl BlockManager {
     pub fn new() -> Self {
         BlockManager::default()
     }
 
+    fn transaction_index_contains(
+        &self,
+        index: &IndexedBlockStore,
+        block_id: &str,
+        id: &str,
+    ) -> Result<bool, BlockManagerError> {
+        if let Some(block) = TransactionIndex::get_block_by_id(index, id)? {
+            let head = index
+                .get(&[block_id])?
+                .next()
+                .expect("BlockStore updated during transaction index check");
+            if block.block_num <= head.block_num {
```

Comment:
Why not just `Ok(block.block_num <= head.block_num)`

Reply:
Yeah, that seems good.